docs/src/prototypes/dropdowns/AsyncDropdownSearch.tsx

```diff
@@ -34,25 +34,55 @@ const createEntry = (): Entry => ({
 // Prototype Search Page View
 // ----------------------------------------
 class AsyncDropdownSearch extends React.Component<{}, SearchPageState> {
+  private searchTimer: number
+
   state = {
     loading: false,
     searchQuery: '',
     items: [],
     value: [],
   }
 
-  searchTimer: number
+  render() {
+    const { items, loading, searchQuery, value } = this.state
+
+    return (
+      <>
+        <Dropdown
+          fluid
+          items={items}
+          loading={loading}
+          loadingMessage={{
+            content: <Loader label="Loading..." labelPosition="end" size="larger" />,
+          }}
+          multiple
+          onSearchQueryChange={this.handleSearchQueryChange}
+          onSelectedChange={this.handleSelectedChange}
+          placeholder="Try to enter something..."
+          search
+          searchQuery={searchQuery}
+          toggleIndicator={false}
+          value={value}
+        />
+        <Divider />
+        <CodeSnippet mode="json" value={this.state} />
+      </>
+    )
+  }
 
-  handleSelectedChange = (e: React.SyntheticEvent, { searchQuery, value }: DropdownProps) => {
+  private handleSelectedChange = (
+    e: React.SyntheticEvent,
+    { searchQuery, value }: DropdownProps,
+  ) => {
     this.setState({ value: value as Entry[], searchQuery })
   }
 
-  handleSearchQueryChange = (e: React.SyntheticEvent, { searchQuery }: DropdownProps) => {
+  private handleSearchQueryChange = (e: React.SyntheticEvent, { searchQuery }: DropdownProps) => {
     this.setState({ searchQuery })
     this.fetchItems()
   }
 
-  fetchItems = () => {
+  private fetchItems = () => {
```

Comment:
https://github.com/stardust-ui/react/pull/901#discussion_r256900108 from @layershifter :

_What is purpose of these changes? Do we need them?
We don't have an agreement on visibility modifiers so let's revert these changes._

Reply:
@layershifter 

> We don't have an agreement on visibility modifiers so let's revert these changes.

I don't get this part, it's pretty obvious to me that in Typescript class members that are not meant to be used outside the scope of the class are private.

> What is purpose of these changes? Do we need them?

I will revert the changes however because the prototype section does not have the JS version so you might argue the change does not bring any benefit here.

Reply:
agree with both sides here. Actually, if we would be absolutely pedantic here and adhere to the flavour of OO paradigm introduced by TS, we indeed should use scope modifiers for components' methods - to showcase which ones meant to be accessible from outside as part of public API.

At the same time, the main objection around having this burden of scope modifiers was reasonable fact that this is pretty rare scenario (that, almost for sure would be invalid in Stardust's case) when we would need to access methods of instantiated React component from outside world. Given this reasonable fact, it was considered to be unnecessary to have this scope modifiers being provided to React components.

That being said, just for the sake of moving forward, lets stick to the thing we've agreed upon (i.e. not introduce scope modifiers for React components) - although, as @Bugaa92 has noticed, this is the prototype's code where this is reasonable approach. And, if there is still strong objection against that (or, for certain, if there is any use case that showcases importance of having these modifiers) - @Bugaa92, please, feel free to schedule some time to discuss that 👍

Reply:
Personally for me: there is no big difference, but I _expect_ to see handlers and helpers method before `render()` methods as it's done in React docs: https://reactjs.org/docs/handling-events.html

Reply:
let's close for now and stick to what we have, tracking in #933